compositor_render/src/wgpu/shader.rs

```diff
@@ -162,7 +162,7 @@ impl WgpuShader {
         });
 
         let common_shader_params =
-            CommonShaderParameters::new(pts, texture_views.len() as u32, target.resolution());
+            CommonShaderParameters::new(pts, textures.len() as u32, target.resolution());
```

Comment:
from what I see original code had `sources.len()`, any reason why you are changing that to textures?

Reply:
This function accepts `textures` instead of `sources`, since it's `pub` and can be used separately (without `node` abstraction over it), making it possible to use in the web renderer :)
```rust
sources: &[(&NodeId, &NodeTexture)]
```

```rust
let textures = sources
            .iter()
            .map(|(_, node_texture)| node_texture.state())
            .collect::<Vec<_>>();
```

Reply:
I missed that texture now is argument(probably misread the diff from the original PR)